hawkular-command-gateway/hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusEndpointProcessors.java

```diff
@@ -0,0 +1,282 @@
+                BusCommand<BasicMessage> command = (BusCommand<BasicMessage>) commands
+                        .getCommand(messageWithData.getBasicMessage().getClass());
+                BusCommandContext context = commandContextFactory.newCommandContext(null);
+                command.execute(messageWithData, context);
+            } catch (Exception e) {
+                log.errorCouldNotProcessBusMessage(basicMessage.getClass().getName(),
+                        messageWithData.getBinaryData() != null, endpoint.getName(), e);
+            }
+        }
+    }
+
+    private static final MsgLogger log = GatewayLoggers.getLogger(BusEndpointProcessors.class);
+
+    @Inject
+    private BusCommands busCommands;
+
+    @Inject
+    private BusCommandContextFactory commandContextFactory;
+
+    @Resource(name = Constants.CONNECTION_FACTORY_JNDI)
+    private ConnectionFactory connectionFactory;
+
+    private BiFunction<String, Session, WsSessionListener> feedSessionListenerProducer;
+    private BiFunction<String, Session, WsSessionListener> uiClientSessionListenerProducer;
+
+    @Inject
+    private WsEndpoints wsEndpoints;
+
+    public void destroy(@Observes @Destroyed(ApplicationScoped.class) Object ignore) {
+        log.debugf("Destroying [%s]", this.getClass().getName());
+        if (feedSessionListenerProducer != null) {
+            wsEndpoints.getFeedSessions().removeWsSessionListenerProducer(feedSessionListenerProducer);
+        }
+        if (uiClientSessionListenerProducer != null) {
+            wsEndpoints.getUiClientSessions().removeWsSessionListenerProducer(uiClientSessionListenerProducer);
+        }
+    }
+
+    /**
+     * This creates the bi-function listener-producers that will create listeners which will
+     * create JMS bus listeners for each websocket session that gets created in the future.
+     *
+     * @param ignore unused
+     */
+    public void initialize(@Observes @Initialized(ApplicationScoped.class) Object ignore) {
+        log.debugf("Initializing [%s]", this.getClass().getName());
+        try {
+            feedSessionListenerProducer = new BiFunction<String, Session, WsSessionListener>() {
+                @Override
+                public WsSessionListener apply(String key, Session session) {
+                    // In the future, if we need other queues/topics that need to be listened to, we add them here.
+                    final Endpoint endpoint = Constants.FEED_COMMAND_QUEUE;
+                    BasicMessageListener<BasicMessage> busEndpointListener = new FeedBusEndpointListener(session, key,
+                            endpoint);
+                    return new BusWsSessionListener(Constants.HEADER_FEEDID, key, endpoint, busEndpointListener);
+                }
+            };
+            wsEndpoints.getFeedSessions().addWsSessionListenerProducer(feedSessionListenerProducer);
+
+            uiClientSessionListenerProducer = new BiFunction<String, Session, WsSessionListener>() {
```

Comment:
This can be simplified with lambdas...
```
uiClientSessionListenerProducer = (key, session) -> {
  final Endpoint endpoint = Constants.UI_COMMAND_QUEUE;
  BasicMessageListener<BasicMessage> busEndpointListener = new UiClientBusEndpointListener(commandContextFactory, busCommands, endpoint);
  return new BusWsSessionListener(Constants.HEADER_UICLIENTID, key, endpoint, busEndpointListener);
}
```